x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/ServiceUtils.java

```diff
@@ -335,6 +335,64 @@ public static Integer extractOptionalPositiveInteger(
         return optionalField;
     }
 
+    public static Float extractOptionalFloat(Map<String, Object> map, String settingName) {
+        return ServiceUtils.removeAsType(map, settingName, Float.class);
+    }
+
+    public static Double extractOptionalDoubleInRange(
+        Map<String, Object> map,
+        String settingName,
+        @Nullable Double minValue,
+        @Nullable Double maxValue,
+        String scope,
+        ValidationException validationException
+    ) {
+        int initialValidationErrorCount = validationException.validationErrors().size();
+        var doubleReturn = ServiceUtils.removeAsType(map, settingName, Double.class, validationException);
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            return null;
+        }
+
+        if (doubleReturn != null && minValue != null && doubleReturn < minValue) {
+            validationException.addValidationError(
+                ServiceUtils.mustBeGreaterThanOrEqualNumberErrorMessage(settingName, scope, doubleReturn, minValue)
+            );
+        }
+
+        if (doubleReturn != null && maxValue != null && doubleReturn > maxValue) {
+            validationException.addValidationError(
+                ServiceUtils.mustBeLessThanOrEqualNumberErrorMessage(settingName, scope, doubleReturn, maxValue)
+            );
+        }
+
+        if (validationException.validationErrors().size() > initialValidationErrorCount) {
+            return null;
+        }
+
+        return doubleReturn;
+    }
+
+    public static <E extends Enum<E>> E extractRequiredEnum(
+        Map<String, Object> map,
+        String settingName,
+        String scope,
+        EnumConstructor<E> constructor,
+        EnumSet<E> validValues,
+        ValidationException validationException
+    ) {
+        var enumReturn = extractOptionalEnum(map, settingName, scope, constructor, validValues, validationException);
+        if (validationException.validationErrors().isEmpty() == false) {
```

Comment:
nit: use the same `int initialValidationErrorCount = validationException.validationErrors().size();` pattern as elsewhere in case the validation error comes from a previous call